### ISSUE TITLE
Regression: Coroutine result with `Unknown | None` resolved before await

### ISSUE THREAD

**Issue by last-partizan** [microsoft/pyright#5705]
**Describe the bug**
pyright@1.1.322 resolves coroutines to it's result before awaiting.

It happens only when result is `Unknown | None`, when i add proper typing for result or input argument - error goes away.

**Code or Screenshots**

```python
async def call(event):
    return event or None

async def main():
    x = call("")
    reveal_type(x)
    await x
```

** Correct result**
```
npx pyright@1.1.320 /tmp/test.py
/tmp/test.py
  /tmp/test.py:6:17 - information: Type of "x" is "Coroutine[Any, Any, Unknown | None]"
```

**Incorrect result**
```
npx pyright@1.1.322 /tmp/test.py
/tmp/test.py
  /tmp/test.py:6:17 - information: Type of "x" is "None"
  /tmp/test.py:7:11 - error: "None" is not awaitable (reportGeneralTypeIssues)
```

**VS Code extension or command-line**

- pyright@1.1.322
- pyright@1.1.321

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression. This will be fixed in the next release. I've added tests to cover this case to avoid any future regressions of this nature.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.323, which I just published. It will also be included in a future release of pylance.
